File: backend/CleanArchitectureApp.Domain/Entities/Project.cs
namespace CleanArchitectureApp.Domain.Entities;

/// <summary>
/// Represents a construction project with multiple phases.
/// Parent entity for all project-related activities.
/// </summary>
public class Project : TenantEntity
{
    public string Id { get; set; } = string.Empty;
    public required string ProjectCode { get; set; } // Unique project identifier
    public required string Name { get; set; }
    public string? Description { get; set; }
    
    // Relationships
    public string ClientId { get; set; } = string.Empty;
    public Client? Client { get; set; }
    
    public string? SiteId { get; set; }
    public Site? Site { get; set; }
    

    
    // Project Manager
    public string? ProjectManagerId { get; set; }
    public User? ProjectManager { get; set; }
    
    // Project Details
    public string ProjectType { get; set; } = "Residential"; // Residential, Commercial, Industrial, Infrastructure
    public decimal? EstimatedBudget { get; set; }
    public decimal? ActualCost { get; set; }
    
    // Timeline
    public DateTime? StartDate { get; set; }
    public DateTime? PlannedEndDate { get; set; }
    public DateTime? ActualEndDate { get; set; }
    
    // Status
    public string Status { get; set; } = "Planning"; // Planning, InProgress, OnHold, Completed, Cancelled
    public int ProgressPercentage { get; set; } = 0;
    
    // Contract Details
    public string? ContractType { get; set; } // Fixed, TimeAndMaterial, CostPlus
    public decimal? ContractValue { get; set; }
    
    public bool IsActive { get; set; } = true;
    
    // Audit fields
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
    
    // Navigation properties
    public ICollection<ProjectPhase> Phases { get; set; } = new List<ProjectPhase>();
    public ICollection<Quotation> Quotations { get; set; } = new List<Quotation>();
    public ICollection<ProjectTask> Tasks { get; set; } = new List<ProjectTask>();
}

File: frontend/contractor-babu-app/src/types/project.ts
export interface Project {
    id: string;
    projectCode: string;
    name: string;
    description?: string;
    clientId: string;
    clientName?: string;
    siteId?: string;
    siteName?: string;
    projectManagerId?: string;
    projectManagerName?: string;
    projectType: string;
    estimatedBudget?: number;
    actualCost?: number;
    startDate?: string;
    plannedEndDate?: string;
    actualEndDate?: string;
    status: 'Planning' | 'InProgress' | 'OnHold' | 'Completed' | 'Cancelled';
    progressPercentage: number;
    contractType?: string;
    contractValue?: number;
    isActive: boolean;
    createdAt: string;
    updatedAt?: string;
}

export interface CreateProjectDto {
    projectCode: string;
    name: string;
    description?: string;
    clientId: string;
    projectManagerId?: string;
    projectType: string;
    estimatedBudget?: number;
    startDate?: string;
    plannedEndDate?: string;
    contractType?: string;
    contractValue?: number;
}

export interface UpdateProjectDto {
    name?: string;
    description?: string;
    projectManagerId?: string;
    projectType?: string;
    estimatedBudget?: number;
    actualCost?: number;
    startDate?: string;
    plannedEndDate?: string;
    actualEndDate?: string;
    status?: string;
    progressPercentage?: number;
    contractType?: string;
    contractValue?: number;
    isActive?: boolean;
}
